tests/formValidation.spec.js

```diff
@@ -0,0 +1,331 @@
+jest.mock('@degjs/dom-utils', () => ({
+    emptyElements: jest.fn(),
+    isElement: jest.fn(o => o.type === 'element')
+}));
+jest.mock('@degjs/object-utils', () => ({
+    ensureArray: jest.fn(item => item)
+}));
+
+jest.mock('../src/utils/state.js');
+jest.mock('../src/utils/idUtils', () => ({
+    getUniqueId: jest.fn()
+}));
+jest.mock('../src/utils/formUtils', () => ({
+    checkFormIntegrity: jest.fn(),
+    checkFieldIntegrity: jest.fn(),
+    checkRuleIntegrity: jest.fn(),
+    disableBrowserValidation: jest.fn()
+}));
+jest.mock('../src/utils/errorUtils', () => ({
+    logError: jest.fn(),
+    scrollToError: jest.fn()
+}));
+
+jest.mock('../src/renderers/renderers', () => ({
+    renderErrorsEl: jest.fn(),
+    renderValidationMessage: jest.fn()
+}));
+
+import formValidation from '../src/formValidation';
+import formUtils from '../src/utils/formUtils';
+import idUtils from '../src/utils/idUtils';
+import errorUtils from '../src/utils/errorUtils';
+import state from '../src/utils/state';
+import {emptyElements} from '@degjs/dom-utils';
+
+describe('formValidation', () => {
+    let formEl;
+    let stateInst;
+
+    beforeEach(() => {
+        formEl = document.createElement('form');
+        const fieldEl = document.createElement('div');
+        fieldEl.classList.add('js-validation-field');
+        const inputEl = document.createElement('input');
+        
+        fieldEl.appendChild(inputEl);
+        formEl.appendChild(fieldEl);
+
+        stateInst = state();
+    });
+
+    afterEach(() => {
+        jest.restoreAllMocks();
+    });
+
+    describe('init', () => {
+       
+        it('should checkFormIntegrity', () => {
+            formValidation(formEl, {});
+            expect(formUtils.checkFormIntegrity).toHaveBeenCalled();
+        });
+
+        it('should disable browser validation', () => {
+            formValidation(formEl, {});
+            expect(formUtils.disableBrowserValidation).toHaveBeenCalled();
+        });
+
+        describe('registerFields', () => {
```

Comment:
Do we need to explicitly test the exposed addFields() method, or do you feel like it's covered by the internal call to registerFields()?

Reply:
Since I have a test suite dedicated to registerFields, I feel that is enough to cover addFields